api/app/routers.py

```diff
@@ -1,15 +1,70 @@
+import logging
 import random
 
 from django.conf import settings
+from django.core.cache import cache
+from django.db import connections
+
+logger = logging.getLogger(__name__)
+CONNECTION_CHECK_CACHE_TTL = 2
+
+
+def connection_check(database: str) -> bool:
+    try:
+        conn = connections.create_connection(database)
+        conn.connect()
+        usable = conn.is_usable()
+        if not usable:
+            logger.warning(
+                f"Unable to access database {database} during connection check"
+            )
+    except Exception:
+        usable = False
+        logger.error(
+            "Encountered exception during connection",
+            exc_info=True,
+        )
+
+    if usable:
+        cache.set(
+            f"db_connection_active.{database}", "online", CONNECTION_CHECK_CACHE_TTL
+        )
```

Comment:
Interesting, I'm not sure I've seen the `cache` used directly like this, I've always used `my_cache = caches[<cache name>]` then called the methods on `my_cache`. How does the core cache work?

Reply:
Really? I've used the `cache` directly most of the time unless I need something specific. 

It's just the default cache with the normal key-value-ttl arguments.

https://docs.djangoproject.com/en/5.0/topics/cache/#django.core.cache.cache